xt/check-signatures.t

```diff
@@ -0,0 +1,146 @@
+    my $type-name = S/'doc/Type/'[(\w+)'/'?]+'.pod6'/$0.join('::')/ with $file;
+    my $type = ::($type-name);
+    CATCH { default { skip "$type-name lacks required introspection" } }
+    TypeDocumentation.parse($file.IO.slurp);
+
+    subtest "check $type-name methods", {
+        plan +$<method>;
+        for $<method> { given .<signature-line> -> $line {
+            use MONKEY-SEE-NO-EVAL;
+            my $method-name = "$type-name\.{$line<name>}";
+            CATCH {
+                when X::AdHoc && .payload ~~ /'::?'/ { skip "$method-name defined with compile-time variable"; }
+                default { skip "cannot check $method-name"; }}
+
+            my $test-msg := "$type-name\.{$line<name>} invocant matches source";
+            my @params = EVAL(":{$line<signature>}").params;
+
+            my $method = do given $line<multi> {
+                when !$type.^lookup($line<name>) { skip "$method-name not found on $type-name";
+                                                   next }
+                when .not { $type.^lookup($line<name>) };
+                given $type.^lookup($line<name>).candidates.grep({.package.^name eq $type-name}) {
+                    when .elems == 1 { .[0] }
+                    default          { skip "cannot determine source for multi method $method-name";
+                                       next }
+            }};
+            my $src-line := slurp( $*PROGRAM.parent(3).add("rakudo/" ~ $method.file.split('::')[1])
+                                 ).lines[ $method.line - 1];
+
+            given $src-line ~~ /<signature>/ {
+                my $src-sig =  do {
+                    when Nil && $src-line ~~ /method/ { '()' }
+                    when Nil  { Nil }
+                    default   { $_ }
+                };
+                when $src-sig ~~ Nil {
+                    when @params[0] eqv Any { flunk($test-msg);
+                                              report-accessor-error($line<signature>, :$src-line)}
+                    unless ok(@params[0].invocant, $test-msg) { report-accessor-error($line<signature> :$src-line) }
+                }
+                when $src-sig ~~ /^'()'/ | /'(-->'/ { pass($test-msg)}
+                when EVAL(":$_").params[0].invocant {
+                    my $is-invocant = @params[0] ?? @params[0].invocant !! False;
+                    unless ok($is-invocant, $test-msg) {
+                        diag("documented method: {$line<signature>}");
+                        diag("    source method: $_");}
+                }
+                default { pass $test-msg }
+            };
+    }}
+}}
+
+sub report-accessor-error($sig, Str :$src-line) {
+    diag("auto-generated accessors must have definite invocants");
+    diag($src-line);
+    diag("documented method: $sig")
+}
+
+run <git checkout ->, :out, :err;
+
+grammar TypeDocumentation {
+    token TOP { [<method> || <other-line>]+ }
+
+    token method { ^^ '=head2 method' \N* \n
```

Comment:
Besides method, other declarations can also appear here: sub, infix, possibly prefix and postcircumfix. Those are, at least, the ones that are recognized by the indexing routine. I see that "other line" can also be recognized, but on that case I'm not sure the actual name of the sub/whatever will be recognized as such.